tests/www/test_security.py

```diff
@@ -99,14 +99,33 @@ def setUp(self):
         log.debug("Complete setup!")
 
     @classmethod
-    def delete_roles(cls):
+    def tearDownClass(cls):
+        cls.delete_test_roles()
+        cls.delete_test_users()
+
+    @classmethod
+    def delete_test_users(cls):
+        method_users = [attr[5:] for attr in cls.__dict__ if attr[:4] == 'test']
+        adhoc_users = [
+            'ElUser',
+            'Monsieur User',
+            'dag_access_user',
+            'LaUser',
+            'dag_permission_user'
+        ]
```

Comment:
Hmm, this fixed list makes me a bit uncomfortable. Why are those `adhoc_users` special? And is there a way to get `method_users` that does not involve magically inferring from name?

Reply:
Adhoc users list is static and taken from this suite's test cases. They are created and never deleted. This func has the same logic (with static list) as existing delete_roles func. So I thought it makes them inline anyway.
Method users are there as there are tests that uses func name as a username, so instead of listing them all in here func tries to find those as well for deletion.
Ideally it would be better to have something like create_user_scope function like I have added in the other PR but that suite uses pytest api.

Reply:
Here #18653

Reply:
I was thinking about the same thing, this should use pytest fixtures to put creation and deletion in one place instead of listing the usernames twice. I'm probably missing some context why this PR is needed if you intend to convert the tests anyway and will leave this for others to comment on.

Reply:
This is just an incremental change to fix tidy up issue that left by the test. But if it's better to refactoring of whole suite to pytest then it's fine, I can close this one can look into the refactoring and fixing it for the good. I was just trying to avoid big changes and not sure how community likes it or not. I can add a new suite with similar name with goal to phase out this one

Reply:
I agree with @uranusjr  - let's refactor it to pytest in one go as well. We are transitoning towards pytest fixtures now and it would be great to use it as an opportunity to do it here as well.

Reply:
Ok, no worries, I will refactor